app/controllers/Login.scala

```diff
@@ -1,12 +1,18 @@
 package controllers
 
-import com.gu.googleauth.{GoogleAuthConfig, LoginSupport}
+import com.gu.googleauth.{GoogleAuthConfig, GoogleGroupChecker, LoginSupport}
 import play.api.libs.ws.WSClient
 import play.api.mvc._
 
 import scala.concurrent.ExecutionContext
 
-class Login(val authConfig: GoogleAuthConfig, val wsClient: WSClient, components: ControllerComponents)(implicit executionContext: ExecutionContext)
+class Login(
+  val authConfig: GoogleAuthConfig,
+  val wsClient: WSClient,
+  requiredGoogleGroups: Set[String],
+  googleGroupChecker: GoogleGroupChecker,
+  components: ControllerComponents
+)(implicit executionContext: ExecutionContext)
```

Comment:
this class has a lot of deps, and the two new ones are only used by a single method.  It would make more sense to separate as necessary so in general things only get what they need.
We could have a Login class with everything else, and a OAuth2 class with a def callback.

not essential for the PR as it won't break anything though!